src/emulator/kernel/include/kernel/state.h

```diff
@@ -59,31 +61,75 @@ struct WaitingThreadData {
     }
 };
 
+// Waiting threads for condition variables don't need a lock count (it's binary)
+struct WaitingThreadDataCondvar {
```

Comment:
I'm using unions inside WaitingThreadData for events. Just a suggestion

Reply:
Not sure how that would work. I need to pass this info at _signal_ call site (ex. [here](https://github.com/Vita3K/Vita3K/pull/236/files#diff-c487eaaba136f04f34b9c19061f96493R275)) whereas this struct is created on the _wait_ function.